src/controllers/articleController.js

```diff
@@ -81,15 +81,75 @@ class ArticleController {
         ]
       });
       if (!articles.length) {
-        return res.status(200).json({
+        return res.status(404).json({
           message: 'No articles found at the moment! please come back later'
         });
       }
       res.status(200).json({
         articles
       });
     } catch (err) {
-      throw (err);
+      throw err;
+    }
+  }
+
+  /**
+   * @description update article
+   * @param {object} req
+   * @param {object} res
+   * @return {object} return object with updated article.
+   */
+  static async updateArticle(req, res) {
+    try {
+      const article = {};
+      const {
+        title, description, body, tags, category
+      } = req.body;
+      const originalArticle = await Article.findOne({ where: { slug: req.userData.slug } });
+      if (title) {
+        article.title = title.trim();
+        article.slug = slugGen(title.trim());
+      }
+      article.description = (description || originalArticle.description).trim();
+      article.body = (body || originalArticle.body).trim();
+      article.tagList = (tags || originalArticle.tagList.toString()).trim().split(/[ ,]+/);
+      article.category = (category || originalArticle.category).trim();
+      if (req.files.image) {
+        article.images = await uploadImage(req.files.image);
+      }
+      const updatedArticle = await Article.update(
+        article,
+        { where: { slug: req.userData.slug }, returning: true, plain: true }
+      );
+      if (updatedArticle) {
+        return res.status(200).json({
+          message: 'Profile updated successfully',
+          article: updatedArticle[1]
+        });
+      }
+    } catch (err) {
+      throw err;
+    }
+  }
+
+  /**
+   * @description delete article
+   * @param {object} req
+   * @param {object} res
+   * @return {object} return message of success or error.
+   */
+  static async deteleArticle(req, res) {
+    try {
+      const deleted = await Article.destroy({
+        where: { slug: req.userData.slug }
```

Comment:
Why did you use slug instead of ID to delete article?

Reply:
We already had slug from the params, I thought it would be efficient to delete using it rather than going to the DB search for an article using slug so as to get the id. But I have got another way of getting the id which I have implemented it.